### PR TITLE
sdk/metric: Remove Reader.ForceFlush and ManualReader.ForceFlush

### PR DESCRIPTION
**Pull request by pellared**
## Why

The `Reader.ForceFlush` and `ManualReader.ForceFlush` are not needed and we can remove it to make the user-facing API cleaner. E.g. we do not need the [Prometheus exporter](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/prometheus) to have `ForceFlush` (via type embedding of `metric.Reader`). `ForceFlush` is only required when periodic reader is used. This change is now possible after #4244.

The [OTel Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md) does NOT require to require to have`Reader.ForceFlush`. For instance the OTel .NET SDK readers do not implement `ForceFlush` at all.

Related PRs and issues in specification:

- https://github.com/open-telemetry/opentelemetry-specification/pull/3563
- https://github.com/open-telemetry/opentelemetry-specification/pull/3609

## What

Remove `Reader.ForceFlush` and `ManualReader.ForceFlush`. Use type assertions so that `ForceFlush` is properly handled by the `MetricProvider`.

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]
LGTM pending the acceptance at the specification level.

**Comment by pellared** [comment]
There is no consensus on specification PR, but I looks like all objections are NOT related to changes proposed in this PR. E.g. see: https://github.com/open-telemetry/opentelemetry-specification/pull/3609#issuecomment-1656214282

**Comment by pellared** [comment]
Now blocked on https://github.com/open-telemetry/opentelemetry-specification/pull/3563

**Review by hanyuancheung — APPROVED** [review]

**Comment by pellared** [comment]
https://github.com/open-telemetry/opentelemetry-specification/pull/3563 is merged 🎉 
This PR can be merged.

File: sdk/metric/config.go
Comment (pellared) [inline sdk/metric/config.go:42]: Should we document this functionality e.g. by adding following comment

```
// ForceFlush calls ForceFlush(context.Context) error
// on all Readers that implements this method.
```

to `func (mp *MeterProvider) ForceFlush(ctx context.Context) error {` ?

File: CHANGELOG.md
Comment (MrAlias) [inline CHANGELOG.md:50]:
```suggestion
- Remove `Reader.ForceFlush` in `go.opentelemetry.io/otel/metric`.
  Notice that `PeriodicReader.ForceFlush` is still available. (#4375)
```

File: sdk/metric/config.go
Comment (MrAlias) [reply sdk/metric/config.go:42]: LGTM

**Review by dashpole — APPROVED** [review]

**Pull request merged**
